### ISSUE TITLE
[BUG] cluster of one fails first time on new install, have to start again

### ISSUE THREAD

**Issue by barentine** [python-microscopy/python-microscopy#391]
**Describe the bug*
Log tree isn't created and the rule server freaks out about it. Running again immediately after is fine

```
(base) ubuntu@greenback:~$ PYMEClusterOfOne

Gtk-Message: 14:43:21.334: Failed to load module "canberra-gtk-module"

Setting icon

INFO:PYME.cluster.cluster_of_one:Launching data server: root=/home/ubuntu/PYMEData

INFO:PYME.cluster.cluster_of_one:Launching rule server

Traceback (most recent call last):

  File "/home/ubuntu/miniconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main

    "__main__", mod_spec)

  File "/home/ubuntu/miniconda3/lib/python3.7/runpy.py", line 85, in _run_code

    exec(code, run_globals)

  File "/home/ubuntu/code/python-microscopy/PYME/cluster/PYMERuleServer.py", line 97, in <module>

    main()

  File "/home/ubuntu/code/python-microscopy/PYME/cluster/PYMERuleServer.py", line 60, in main

    dist_err_handler = logging.handlers.RotatingFileHandler(filename=dist_log_err_file, mode='w', maxBytes=1e6, backupCount=1)

  File "/home/ubuntu/miniconda3/lib/python3.7/logging/handlers.py", line 148, in __init__

    BaseRotatingHandler.__init__(self, filename, mode, encoding, delay)

  File "/home/ubuntu/miniconda3/lib/python3.7/logging/handlers.py", line 55, in __init__

    logging.FileHandler.__init__(self, filename, mode, encoding, delay)

  File "/home/ubuntu/miniconda3/lib/python3.7/logging/__init__.py", line 1087, in __init__

    StreamHandler.__init__(self, self._open())

  File "/home/ubuntu/miniconda3/lib/python3.7/logging/__init__.py", line 1116, in _open

    return open(self.baseFilename, self.mode, encoding=self.encoding)

FileNotFoundError: [Errno 2] No such file or directory: '/home/ubuntu/PYMEData/LOGS/distributor.log'

INFO:numexpr.utils:Note: NumExpr detected 12 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.

INFO:numexpr.utils:NumExpr defaulting to 8 threads.

INFO:root:========================================

PYMEDataServer, running on python 3.7.7 (default, May  7 2020, 21:25:33) 

[GCC 7.3.0]

INFO:root:Serving from directory: /home/ubuntu/PYMEData

INFO:root:Serving HTTP on 127.0.0.1 port 33209 ...

INFO:PYME.cluster.cluster_of_one:Launching node server

INFO:PYME.cluster.cluster_of_one:Shutting down cluster

Shutting down cluster

dict_values([])

Traceback (most recent call last):

  File "/home/ubuntu/miniconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main

    "__main__", mod_spec)

  File "/home/ubuntu/miniconda3/lib/python3.7/runpy.py", line 85, in _run_code

    exec(code, run_globals)

  File "/home/ubuntu/code/python-microscopy/PYME/cluster/PYMERuleNodeServer.py", line 165, in <module>

    main()

  File "/home/ubuntu/code/python-microscopy/PYME/cluster/PYMERuleNodeServer.py", line 98, in main

    proc = rulenodeserver.ServerThread(distributors[0], serverPort, externalAddr=externalAddr, profile=False)

IndexError: list index out of range

Traceback (most recent call last):

  File "/home/ubuntu/miniconda3/bin/PYMEClusterOfOne", line 33, in <module>

    sys.exit(load_entry_point('python-microscopy', 'console_scripts', 'PYMEClusterOfOne')())

  File "/home/ubuntu/code/python-microscopy/PYME/cluster/cluster_of_one.py", line 178, in main

    cluster.launch(gui=options.ui, clusterUI=options.clusterui)

  File "/home/ubuntu/code/python-microscopy/PYME/cluster/cluster_of_one.py", line 100, in launch

    self._launch_ruleserver_ui()

  File "/home/ubuntu/code/python-microscopy/PYME/cluster/cluster_of_one.py", line 77, in _launch_ruleserver_ui

    webbrowser.open_new_tab(list(ruleservers.values())[0])

IndexError: list index out of range

(base) ubuntu@greenback:~$ PYMEClusterOfOne

Gtk-Message: 14:43:34.552: Failed to load module "canberra-gtk-module"

Setting icon

INFO:PYME.cluster.cluster_of_one:Launching data server: root=/home/ubuntu/PYMEData

INFO:PYME.cluster.cluster_of_one:Launching rule server

INFO:ruleserver:Starting ruleserver on 127.0.0.1:0

INFO:numexpr.utils:Note: NumExpr detected 12 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.

INFO:numexpr.utils:NumExpr defaulting to 8 threads.

INFO:root:========================================

PYMEDataServer, running on python 3.7.7 (default, May  7 2020, 21:25:33) 

[GCC 7.3.0]

INFO:root:Serving from directory: /home/ubuntu/PYMEData

INFO:root:Serving HTTP on 127.0.0.1 port 32811 ...

INFO:PYME.cluster.cluster_of_one:Launching node server

dict_values(['http://127.0.0.1:35195/'])

DEBUG:nodeserver:Starting to poll for tasks

nodeserver:INFO:Starting nodeserver on 127.0.0.1:0

INFO:nodeserver:Starting nodeserver on 127.0.0.1:0

127.0.0.1 - - [04/Aug/2020 14:43:40] code 404, message No handler for /favicon.ico

DEBUG:nodeserver:Launching worker processors
```

**To Reproduce**
1. remove all PYME folders ever
2. `PYMEClusterOfOne`

**Expected behavior**
make the folders necessary the first time

**Platform (please complete the following information - or copy-paste from error dialog if shown):**
 - OS: [e.g. Win10]
ubuntu 18.04
 - Python version:
3.7
 - numpy version:
1.19
 - wxpython version [if a UI bug]:
4.0.4

**additional context**
I guess I cheated my manually creating a PYMEData directory first, haven't looked yet to see if that would change things

**Comment by barentine** [comment]
ran cluster of one on my instrument computer for the first time today and also ran into this. Fails first run, works fine the second
```

(pyme) C:\Users\Bergamot>PYMEClusterOfOne
Setting icon
INFO:PYME.cluster.cluster_of_one:Launching data server: root=C:\Users\Bergamot\PYMEData
INFO:PYME.cluster.cluster_of_one:Launching rule server
Traceback (most recent call last):
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "c:\users\bergamot\code\python-microscopy\PYME\cluster\PYMERuleServer.py", line 97, in <module>
    main()
  File "c:\users\bergamot\code\python-microscopy\PYME\cluster\PYMERuleServer.py", line 60, in main
    dist_err_handler = logging.handlers.RotatingFileHandler(filename=dist_log_err_file, mode='w', maxBytes=1e6, backupCount=1)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\logging\handlers.py", line 150, in __init__
    BaseRotatingHandler.__init__(self, filename, mode, encoding, delay)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\logging\handlers.py", line 57, in __init__
    logging.FileHandler.__init__(self, filename, mode, encoding, delay)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\logging\__init__.py", line 1032, in __init__
    StreamHandler.__init__(self, self._open())
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\logging\__init__.py", line 1061, in _open
    return open(self.baseFilename, self.mode, encoding=self.encoding)
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Bergamot\\PYMEData\\LOGS\\distributor.log'
INFO:numexpr.utils:Note: NumExpr detected 12 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.
INFO:numexpr.utils:NumExpr defaulting to 8 threads.
INFO:root:========================================
PYMEDataServer, running on python 3.6.10 |Anaconda, Inc.| (default, May  7 2020, 19:46:08) [MSC v.1916 64 bit (AMD64)]

INFO:root:Serving from directory: C:\Users\Bergamot\PYMEData
INFO:root:Serving HTTP on 127.0.0.1 port 59269 ...
INFO:PYME.cluster.cluster_of_one:Launching node server
INFO:PYME.cluster.cluster_of_one:Shutting down cluster
Shutting down cluster
dict_values([])
Traceback (most recent call last):
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "c:\users\bergamot\code\python-microscopy\PYME\cluster\PYMERuleNodeServer.py", line 165, in <module>
    main()
  File "c:\users\bergamot\code\python-microscopy\PYME\cluster\PYMERuleNodeServer.py", line 98, in main
    proc = rulenodeserver.ServerThread(distributors[0], serverPort, externalAddr=externalAddr, profile=False)
IndexError: list index out of range
Traceback (most recent call last):
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\Scripts\PYMEClusterOfOne-script.py", line 33, in <module>
    sys.exit(load_entry_point('python-microscopy', 'console_scripts', 'PYMEClusterOfOne')())
  File "c:\users\bergamot\code\python-microscopy\PYME\cluster\cluster_of_one.py", line 178, in main
    cluster.launch(gui=options.ui, clusterUI=options.clusterui)
  File "c:\users\bergamot\code\python-microscopy\PYME\cluster\cluster_of_one.py", line 100, in launch
    self._launch_ruleserver_ui()
  File "c:\users\bergamot\code\python-microscopy\PYME\cluster\cluster_of_one.py", line 77, in _launch_ruleserver_ui
    webbrowser.open_new_tab(list(ruleservers.values())[0])
IndexError: list index out of range

(pyme) C:\Users\Bergamot>PYMEClusterOfOne
Setting icon
INFO:PYME.cluster.cluster_of_one:Launching data server: root=C:\Users\Bergamot\PYMEData
INFO:PYME.cluster.cluster_of_one:Launching rule server
INFO:ruleserver:Starting ruleserver on 127.0.0.1:0
INFO:numexpr.utils:Note: NumExpr detected 12 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.
INFO:numexpr.utils:NumExpr defaulting to 8 threads.
INFO:root:========================================
PYMEDataServer, running on python 3.6.10 |Anaconda, Inc.| (default, May  7 2020, 19:46:08) [MSC v.1916 64 bit (AMD64)]

INFO:root:Serving from directory: C:\Users\Bergamot\PYMEData
INFO:root:Serving HTTP on 127.0.0.1 port 59281 ...
INFO:PYME.cluster.cluster_of_one:Launching node server
dict_values(['http://127.0.0.1:59279/'])
DEBUG:nodeserver:Starting to poll for tasks
nodeserver:INFO:Starting nodeserver on 127.0.0.1:0
INFO:nodeserver:Starting nodeserver on 127.0.0.1:0
DEBUG:nodeserver:Launching worker processors
```

**Comment by barentine** [comment]
Note also you can hit this not through cluster of one, but through e.g. ruleserver queuing analysis
```
$ PYMERuleServer
Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/bin/PYMERuleServer", line 33, in <module>
    sys.exit(load_entry_point('python-microscopy', 'console_scripts', 'PYMERuleServer')())
  File "/home/ubuntu/code/python-microscopy/PYME/cluster/PYMERuleServer.py", line 60, in main
    dist_err_handler = logging.handlers.RotatingFileHandler(filename=dist_log_err_file, mode='w', maxBytes=1e6, backupCount=1)
  File "/home/ubuntu/miniconda3/lib/python3.7/logging/handlers.py", line 148, in __init__
    BaseRotatingHandler.__init__(self, filename, mode, encoding, delay)
  File "/home/ubuntu/miniconda3/lib/python3.7/logging/handlers.py", line 55, in __init__
    logging.FileHandler.__init__(self, filename, mode, encoding, delay)
  File "/home/ubuntu/miniconda3/lib/python3.7/logging/__init__.py", line 1087, in __init__
    StreamHandler.__init__(self, self._open())
  File "/home/ubuntu/miniconda3/lib/python3.7/logging/__init__.py", line 1116, in _open
    return open(self.baseFilename, self.mode, encoding=self.encoding)
FileNotFoundError: [Errno 2] No such file or directory: '/data/PYMEData/LOGS/distributor.log'

```

I'll stop being lazy and have a look/PR about this sometime this week

**Comment by barentine** [comment]
Hopefully fixed with #494. Admittedly haven't tested on a _fresh_ install (so if we're missing another downstream error, I wouldn't know yet), but will reopen if necessary.
